### PR TITLE
Mark exceptions not handled by user as `handled: false`

### LINKED ISSUE #456: Exceptions not handled by user code should be `unhandled: true`
Currently all exceptions are marked as `handled`.
This decision was made because exceptions in Flutter don't crash the app and thus sessions should not be marked as crashed.

Though we want to mark exceptions as unhandled to indicate that an exception caught by the root zone handler was not handled by the user.

There should be a way to differentiate.

This issue is a result of a [discussion on Discord](https://discord.com/channels/621778831602221064/654883298736340994/840233050905903186).

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- Mark exceptions not handled by user as `handled: false`

## :bulb: Motivation and Context

Closes #456

## :green_heart: How did you test it?

Update & run tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Enhancements
```
Comment (marandaneto) [inline CHANGELOG.md]: Probably a `Feature` and I'd add a note about a breaking change that can affect the release health data.
I'm wondering if this requires a major bump even.

**Comment by marandaneto** [comment]
Can we test this on Android and iOS? that the session is marked as `crashed` but just `errored` when it's a handled error (try/catch)? it's important to test on both platforms.

**Comment by denrase** [comment]
# Test Approach on sentry.io

- start app
- trigger event/crash
- put app in background for > 30 seconds
- kill app
- start app
- trigger event/crash

## try/catch iOS 7.8.1 (2) - Bundle ID io.sentry.flutter.example

https://sentry-sdks.sentry.io/issues/3812307661/?project=5428562&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=1h&stream_index=1

https://sentry-sdks.sentry.io/releases/io.sentry.flutter.example%407.8.1%2B2/?chart=sessionCount&pageStatsPeriod=1h&project=5428562

✅ Two issues on sentry.io
✅ Tow errored sessions on sentry.io

## try/catch Android 7.8.1 (2) - Bundle ID io.sentry.samples.flutter

https://sentry-sdks.sentry.io/issues/3812307661/?project=5428562&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=1h&stream_index=0

https://sentry-sdks.sentry.io/releases/io.sentry.samples.flutter%407.8.1%2B2/?chart=sessionCount&pageStatsPeriod=1h&project=5428562

✅ Two issues on sentry.io
✅ Tow errored sessions on sentry.io

## unhandled iOS 7.8.1 (2) - Bundle ID io.sentry.flutter.example

https://sentry-sdks.sentry.io/issues/4293490306/?project=5428562&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=1h&stream_index=1
https://sentry-sdks.sentry.io/releases/io.sentry.flutter.example%407.8.1%2B2/?chart=sessionCount&pageStatsPeriod=1h&project=5428562

✅ Two unhandled issues on sentry.io
❌ No unhandled/crashed session on sentry.io, only errored

## unhandled Android 7.8.1 (2) - Bundle ID io.sentry.samples.flutter

https://sentry-sdks.sentry.io/issues/4293490306/events/27e13d937a0b4144ae1d8b6e1caff772/?project=5428562&query=is%3Aunresolved&referrer=next-event&statsPeriod=1h&stream_index=1

✅ Two unhandled issues on sentry.io
✅ Three unhandled/crashed sessions on sentry.io
	- Three Healthy sessions on sentry.io

**Comment by marandaneto** [comment]
@denrase after reading https://github.com/getsentry/sentry-dart/pull/1535#issuecomment-1620148531 its unclear to me if everything worked as expected or if there's something to be fixed, I recall you mentioned something on iOS, did it work?

**Comment by denrase** [comment]
@marandaneto I updated the comment to make it more clear. I could never get unhandled/crashed sessions to appear on iOS.

**Comment by marandaneto** [comment]
> @marandaneto I updated the comment to make it more clear. I could never get unhandled/crashed sessions to appear on iOS.

Ok, so we have to follow up with @brustolin and check the iOS implementation.

**Comment by brustolin** [comment]
> Ok, so we have to follow up with @brustolin and check the iOS implementation.

This is the implementation https://github.com/getsentry/sentry-cocoa/blob/1c55ca49f313053fcdbf84f7ea2a37e65174ccee/Sources/Sentry/SentryHub.m#L640-L652

**Comment by marandaneto** [comment]
> > Ok, so we have to follow up with @brustolin and check the iOS implementation.
> 
> This is the implementation [getsentry/sentry-cocoa@`1c55ca4`/Sources/Sentry/SentryHub.m#L640-L652](https://github.com/getsentry/sentry-cocoa/blob/1c55ca49f313053fcdbf84f7ea2a37e65174ccee/Sources/Sentry/SentryHub.m#L640-L652)

@denrase let's work together with @brustolin and find out where the problem is.

**Comment by denrase** [comment]
@marandaneto @brustolin Forgot to update the cocoa SDK to the version that had the new code. Now everything works as expected, the unhandled errors are reported as crashed sessions. Sorry for the confusion.

Bildschirmfoto 2023-07-17 um 14 12 33

File: flutter/ios/sentry_flutter.podspec [hunk 2]
```diff
@@ -12,7 +12,7 @@ Sentry SDK for Flutter with support to native through sentry-cocoa.
                          :tag => s.version.to_s }
   s.source_files     = 'Classes/**/*'
   s.public_header_files = 'Classes/**/*.h'
-  s.dependency 'Sentry/HybridSDK', '8.8.0'
+  s.dependency 'Sentry/HybridSDK', '8.9.1'
```
Comment (marandaneto) [inline flutter/ios/sentry_flutter.podspec]: Merge https://github.com/getsentry/sentry-dart/pull/1553/files into your branch or use it as your base branch instead.

**Comment by marandaneto** [comment]
This PR has to be based on the major branch v8.0.0 since it may mess up with all the release health data (crash free sessions/users).

**Comment by ueman** [comment]
I'm really excited for this. Thanks for tackling it

**Pull request merged**